packages/connectivity/tizen/src/connectivity_tizen_plugin.cc

```diff
@@ -0,0 +1,179 @@
+#include "connectivity_tizen_plugin.h"
+
+#include <net_connection.h>
+#include <wifi-manager.h>
+
+#include <flutter/method_channel.h>
+#include <flutter/plugin_registrar.h>
+#include <flutter/standard_method_codec.h>
+#include <flutter/event_channel.h>
+#include <flutter/event_stream_handler_functions.h>
+#include <flutter/event_sink.h>
+
+
+#include <map>
+#include <memory>
+#include <sstream>
+#include <string>
+
+#include "log.h"
+
+class ConnectivityTizenPlugin : public flutter::Plugin {
+ public:
+  static void RegisterWithRegistrar(flutter::PluginRegistrar *registrar) {
+    LOG_INFO("RegisterWithRegistrar");
+    auto method_channel =
+        std::make_unique<flutter::MethodChannel<flutter::EncodableValue>>(
+            registrar->messenger(), "plugins.flutter.io/connectivity",
+            &flutter::StandardMethodCodec::GetInstance());
+    auto event_channel =
+        std::make_unique<flutter::EventChannel<flutter::EncodableValue>>(
+            registrar->messenger(), "plugins.flutter.io/connectivity_status",
+            &flutter::StandardMethodCodec::GetInstance());
+
+    auto plugin = std::make_unique<ConnectivityTizenPlugin>();
+    method_channel->SetMethodCallHandler(
+        [plugin_pointer = plugin.get()](const auto &call, auto result) {
+          plugin_pointer->HandleMethodCall(call, std::move(result));
+        });
+    auto event_channel_handler = std::make_unique<flutter::StreamHandlerFunctions<>>(
+        [plugin_pointer = plugin.get()](const flutter::EncodableValue *arguments,
+                                        std::unique_ptr<flutter::EventSink<>> &&events)
+            -> std::unique_ptr<flutter::StreamHandlerError<>> {
+          LOG_INFO("OnListen");
+          plugin_pointer->registerObsever(std::move(events));
+          return nullptr;
+        },
+        [plugin_pointer = plugin.get()](const flutter::EncodableValue *arguments)
+            -> std::unique_ptr<flutter::StreamHandlerError<>> {
+          LOG_INFO("OnCancel");
+          plugin_pointer->clearObserver();
+          return nullptr;
+        });
+    event_channel->SetStreamHandler(std::move(event_channel_handler));
+    registrar->AddPlugin(std::move(plugin));
+  }
+
+  ConnectivityTizenPlugin()
+      : m_connection(nullptr), m_events(nullptr)
+  {
+    ensureConnectionHandle();
+  }
+
+  virtual ~ConnectivityTizenPlugin() 
+  {
+    if (m_connection != nullptr)
+    {
+      connection_destroy(m_connection);
+      m_connection = nullptr;
+    }
+  }
+
+  void registerObsever(std::unique_ptr<flutter::EventSink<flutter::EncodableValue>> &&events)
+  {
+    ensureConnectionHandle();
+    if (connection_set_type_changed_cb(m_connection, connetionTypeChangedCB, this) != CONNECTION_ERROR_NONE)
```

Comment:
I am not sure but did you set clang_format_style as Google?

Reply:
done.